### DISCUSSION TITLE
3.0

### DISCUSSION THREAD

**Discussion by filecxx** [filecxx/FileCentipede#402] [Ideas]
**Plugins & Script**
**File search engine**
ED2K
File browser - WebDrive 
Socket5 proxy server
Proxy spider
DHT spider
Web spider
QR code
Open with...
Drop Basket
Speed monitor
Usenet
Premium link generator
....
....
....
...

**Comment by lzymbz** [comment]
支持，每天都会到这里溜达一圈，就看有没有啥更新

**Comment by cesaryuan** [comment]
建议：
1. 保留已完成项目的平均下载速度信息
2. #411

**Comment by GATOQSECOMIO** [comment]
**Download/P2P Protocols**
+ + **KAD / Kademlia (Emule)** protocol implemented or any like this (with capability to search on the compatible clients with this protocol)? (like [aMule](https://github.com/amule-project/amule) & [eMule](https://github.com/irwir/eMule))
+ + **Webtorrent** peers compatibility.
+ + Add **Soulseek** Protocol (like [Nicotine+](https://github.com/nicotine-plus/nicotine-plus))
+ + Add **Gnutella & Gnutella2** protocols & **Direct Connect** (like [GTK-Gnutella](https://github.com/gtk-gnutella/gtk-gnutella) & [ivan386/Shareaza](https://github.com/ivan386/Shareaza) & [ansani/Shareaza](https://github.com/ansani/Shareaza))
+ + IRC XDCC support ([xdcc-grabscher](https://github.com/lformella/xdcc-grabscher) & [RecklessXDCC](https://github.com/coderiekelt/RecklessXDCC) & [xddcwebloader](https://github.com/MaggiWuerze/xddcwebloader) & [xdcc-dl](https://github.com/namboy94/xdcc-dl))
+ + Capability to active or disable plugins & scripts

Thanks & Congrats for the Project!

**Reply by GATOQSECOMIO** [reply]
+ Dark mode
+ Pre-allocation of torrent files on disk
+ Ability for add or remove columns in download/upload list.
+ Tags

**Reply by c-sanchez** [reply]
+1 for dark mode

**Comment by whxuezhe** [comment]
希望3.0支持批量磁力链接任务添加，一次可以下载6集剧集的功能。还有就是3.0大工程的进度询问，国庆前有望发版吗？

**Reply by filecxx** [reply]
有

**Comment by trsl2000** [comment]
2.8.2版本，webdav连接后，目录结构按名称排序时会错乱。

**Comment by NebulaTechs** [comment]
什么时候发布？

**Comment by Terkyz** [comment]
Group of downloads: In Jdownloader, for example, I have the ability to define within the download manager itself that a whole group of links are equivalent to the same download, it has no other function than the organization.

**Comment by c-sanchez** [comment]
Option to choose which columns to show/hide.

**Comment by Terkyz** [comment]
When downloading any file in Linux (at least KDE Plasma) with Firefox, the "Add task" menu should appear and not Dolphin, since it is redudant and then the Add Task menu appears.

It would also be great to be able to choose in path on Dolphin and have the Add task menu automatically select that path.

**Comment by Terkyz** [comment]
Hover over the tray icon to see the current total download/upload speed.

**Comment by Terkyz** [comment]
It would be great to be able to automatically transcribe the format of a multimedia file when downloading it.

For example, suppose I want to download a video but it uses the .mkv format and I want it in .mp4 (for X or Y reasons), it would be great if File Centipede would ask the user if he wants to transcribe it to other formats.

And in case of images, I personally hate webp that usually don't have preview or are not compatible with certain software, it would be great if File Centipede could automatically convert it to .png/.jpg according to the user's request.

**Comment by jobswelldone** [comment]
1. option to  auto sort file in diffrent folder according to there  extensions
2.  options to add custom themes
3. option to  import  proxy from a txt/json/xml file or from a url .
4. option to use multiple  proxy to  overcome soft dowload limit 
5. pulgin api / Scripting api (will reduce the work you need to do )
6. force use randomly genratored password for security purpose

**Comment by zrq0058** [comment]
建议：
添加下载完成自动关机功能

**Comment by lordshenone** [comment]
希望能出个docker版，装在NAS上，通过网页或客户端来添加下载任务。

**Reply by badrelmers** [reply]
I hope we can have an option to disable torrent, the app is wonderful but having a torrent client connected all the time that you never use is so bad

**Comment by signin404** [comment]
添加个修改BT监听端口的命令行参数
libtorrent能支持TCP和UDP使用不同的监听端口吗?

好像只改announce_port不改监听端口也行
目前的"Torrent 启用代理"是仅跟踪器代理吗?

**Comment by AmyCoa** [comment]
3.0有消息吗，大概什么时候出啊
